### PR TITLE
Contributing guide update

### LINKED ISSUE #8820: Update Contributing guide
### Summarize Functionality

Update guide for contributors to work with our current structure with V2 release.

Will likely rework some of the content that may not be needed anymore.

### Is there a command that is similiar or close to what you are looking for?

Yes

### Technical Details

_No response_

### PR DESCRIPTION
**Pull request by wsmelton**
Documentation update for contributing guide.

fixes #8820

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+    As of v2 release the dbatools library, SMO library, and 3rd party libaries have been split out into both a separate repository and [PowerShell module](https://powershellgallery.com/packages/dbatools.library).
+
+1. [dataplat/appveyor-lab](https://github.com/dataplat/appveyor-lab)
+
+    We use Appveyor for running the Pester test for each function. We have this repository to store content that is used in our tests. This keeps the current repository clean from excess files just for testing. **Maintainers will determine if tests require files be placed in this repository**.
+
+1. [dataplat/docs](https://github.com/dataplat/docs)
+
+    This repository and content are auto-generated from the comment-based help (CBH) within each command for the module. Any things found wrong in a given command's help content is addressed in the dbatools module's repository.
```
Comment (jpomfret) [inline CONTRIBUTING.md:25]:
```suggestion
    This repository and content are auto-generated from the comment-based help (CBH) within each command for the module. Fixes and changes to the help content for a given command is addressed in the dbatools module's repository.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+1. [dataplat/web](https://github.com/dataplat/web)
+
+    This repository is were the [dbatools.io](https://dbatools.io) site content is hosted.
+
+    > **NOTE** Blog post content **is not** kept updated, it is after all post from scenarios or situations at the time of posting by the author(s).
```
Comment (jpomfret) [inline CONTRIBUTING.md:31]: Not sure this sentence makes sense. Maybe just changing post to posts, or maybe

```suggestion
    > **NOTE** Blog post content **is not** kept updated, they are descriptions of scenarios or situations at the time of posting by the author(s).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+1. [dataplat/docker](https://github.com/dataplat/docker)
+
+    This repository hosts the code to generate the published Docker image for the module.
+
+## Getting Help
+
+Our [dbatools-dev](https://aka.ms/sqlslack) Slack channel is best avenue for asking for assistance or needing review of any work on the module. The main contributors stay active on this channel during the week for the most part.
```
Comment (jpomfret) [inline CONTRIBUTING.md:39]:
```suggestion
Our [dbatools-dev](https://aka.ms/sqlslack) Slack channel is the best avenue for asking for assistance or needing review of any work on the module. The main contributors try and stay active on this channel during the week.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+> _It can be easy to get overwhelmed if you are starting out and have not worked with things like Git, GitHub, and even PowerShell. If you are reading this its a great first step 🍾. We are happy to help you get involved with our project. Please don't hesitate to use the contact methods above to reach out if you want to get involved._
+
+### 🐛 Reports
+
+> Experience: **Any**
+
+This can be anything from finding bugs in our current or preview release that our tests have missed to combing through our current issues to confirm they still exist.
+
+[Open a new issue](https://dbatools.io/new-issue/) GitHub and fill in all the details. The title should report the affected function, followed by a brief description (e.g. _Get-DbaDatabase - Add property x to default view_). The provided template holds the details maintainers and contributors will need in order to confirm whether it is reproducible.
```
Comment (jpomfret) [inline CONTRIBUTING.md:55]:
```suggestion
[Open a new issue](https://dbatools.io/new-issue/) on GitHub and fill in all the details. The title should report the affected function, followed by a brief description (e.g. _Get-DbaDatabase - Add property x to default view_). The provided template holds the details maintainers and contributors will need in order to confirm whether it is reproducible.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+### 📃 Documentation
+
+> Experience: **Beginner** (Git, Markdown, PowerShell)
+
+Contributions in this area can help you get familiar with our commands and some of our common parameters, naming conventions, and using Git and interacting with GitHub Pull Requests (if any of this is new to you).
+
+We strive to have all of our documentation as accurate as possible so a few areas that you can help:
+
+- Examples: Do they actually work? Do they do what the associated description states? Are the parameters still accurate?
+- Grammar: We default to English language in our help documentation. Are there any grammar errors?
```
Comment (jpomfret) [inline CONTRIBUTING.md:66]: Should we specify US English?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+- Anything else: fresh set of eyes on documentation can always find things we may have missed or spot checking for areas that can be improved.
+
+### Fixing 🐛
+
+> **NOTE** Please ensure you have an issue to work on and any discussions on the fix you will implement are done in that issue. This helps keep the background on the change in one place and removes any guesswork.
+
+We have a [step-by-step guide](https://dbatools.io/firstpull) if you don't know Github enough.
```
Comment (jpomfret) [inline CONTRIBUTING.md:73]:
```suggestion
We have a [step-by-step guide](https://dbatools.io/firstpull) if your new to Github or need a refresh.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+1. Use camelCase for multiple word variables (e.g., `$currentLogin`, `$dbLogin`)
+1. Foreach loops should use plural collections and singular variable in the loop (e.g., `foreach ($db in $databases) {...}` or `foreach ($db in $dbs) {...}`)
+
+[This page](https://github.com/dataplat/dbatools/wiki/Standard-Documentation) sums up what we currently use. We aim at standardizing and reducing to a set of self-documenting and reusable variables.
+
+### Formatting and indentation
+
+We 💘 OTBS with this project. Contributors will find an easier experience with our formatting standards using [Visual Studio Code](https://aka.ms/vscode). If for some reason your instance of VS Code is not formatting the PowerShell file you can leveraging `Invoke-DbatoolsFormatter` which does the groundwork for you.
+
+**We do have a tests that will validate the format of the file in any Pull Request.**
```
Comment (jpomfret) [inline CONTRIBUTING.md:105]:
```suggestion
**We do have a pester test that will validate the format of the file in any Pull Request.**
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+- 2016_2017 : a server with two instances available, 2016 and 2017 Developer Edition
+- default: a server with two instances available, one SQL Server 2008 R2 Express Edition and a SQL Server 2016 Developer Edition
+
+Builds are split among "scenario"(s) because not every test requires everything to be up and running, and resources on AppVeyor are constrained.
+
+Ideally:
+
+1. Whenever possible, write UnitTests.
+1. You should write IntegrationTests ideally running in **EITHER** the 2008R2 or the 2016 "scenario".
+1. Default and 216_2017 are the most resource constrained and are left to run the Copy-* commands which are the only ones **needing** two active instances.
```
Comment (jpomfret) [inline CONTRIBUTING.md:143]:
```suggestion
1. Default and 2016_2017 are the most resource constrained and are left to run the Copy-* commands which are the only ones **needing** two active instances.
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+1. If you want to write tests that, e.g, target **BOTH** 2008R2 and 2016, try to avoid writing tests that need both instances to be active at the same time.
+
+AppVeyor is set up to recognize what "scenario" is required by your test, simply inspecting for the presence of combinations of `$script:instance1`, `$script:instance2` and `$script:instance3`. If you need to fall into case (4), write two test files, e.g. _Get-DbaFoo.first.Tests.ps1_ (targeting `$script:instance1` only) and _Get-DbaFoo.second.Tests.ps1_ (targeting `$script:instance2` only).
+
+If you don't want to wait for the entire test suite to run (i.e. you need to run only `Get-DbaFoo` on AppVeyor), you can use a **_magic command_** within the commit message, namely `(do Get-DbaFoo)` . This will run only test files within the test folder matching this mask `tests\*Get-DbaFoo*.Tests.ps1`.
```
Comment (jpomfret) [inline CONTRIBUTING.md:148]: I didn't know that! Very cool

**Review by jpomfret — APPROVED** [review]
Looks great! Added a few suggestions

**Comment by andreasjordan** [comment]
Looks very good, thanks.

**Review by andreasjordan — APPROVED** [review]

File: CONTRIBUTING.md (hunk 9 above)
Comment (wsmelton) [reply CONTRIBUTING.md:148]: You can do this for your branch as well, includes forks.

File: CONTRIBUTING.md (hunk 2 above)
Comment (wsmelton) [reply CONTRIBUTING.md]: Decided to just remove.

**Comment by potatoqualitee** [comment]
boom chaka 💥

**Pull request merged**